src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkWatchListService.java

```diff
@@ -73,27 +83,59 @@ public void removeWatch(Endpoint endpoint) {
         endpoints.remove(endpoint);
     }
 
+    @VisibleForTesting
+    void updateExhaustedPollMode(Endpoint endpoint, boolean enable, long currentTime) {
+        log.info("Discovery poll mode for endpoint {} update - exhausted mode requested (enable: '{}')",
+                endpoint, enable);
+        WatchListEntry watchListEntry = endpoints.computeIfPresent(endpoint,
+                (e, listEntry) -> listEntry.toBuilder().exhaustedPollEnabled(enable).build());
+
+        if (watchListEntry != null && !enable) { // when ISL belonging to this endpoint was found
+            reloadEndpointTimeout(endpoint, currentTime);
+        }
+    }
+
+    public void updateExhaustedPollMode(Endpoint endpoint, boolean enable) {
+        updateExhaustedPollMode(endpoint, enable, now());
+    }
+
+    @VisibleForTesting
+    void updateAuxiliaryPollMode(Endpoint endpoint, boolean enable, long currentTime) {
+        log.info("Discovery poll mode for endpoint {} update - auxiliary mode requested (enable: '{}')",
+                endpoint, enable);
+        WatchListEntry watchListEntry = endpoints.computeIfPresent(endpoint,
+                (e, listEntry) -> listEntry.toBuilder().auxiliaryPollEnabled(enable).build());
+
+        if (watchListEntry != null && !enable) { // when another mechanism used to determine ISL status is disabled
+            reloadEndpointTimeout(endpoint, currentTime);
+        }
+    }
+
+    public void updateAuxiliaryPollMode(Endpoint endpoint, boolean enable) {
+        updateAuxiliaryPollMode(endpoint, enable, now());
+    }
+
     /**
      * Consume timer tick.
      */
-    public void tick(long tickTime) {
+    @VisibleForTesting
+    void tick(long tickTime) {
         SortedMap<Long, Set<Endpoint>> range = timeouts.subMap(0L, tickTime + 1);
         if (!range.isEmpty()) {
-            HashSet<Endpoint> renew = new HashSet<>();
-            for (Set<Endpoint> e : range.values()) {
-                for (Endpoint ee : e) {
-                    if (endpoints.contains(ee)) {
-                        carrier.discoveryRequest(ee, tickTime);
-                        renew.add(ee);
+            Map<Endpoint, WatchListEntry> renew = new HashMap<>();
+
+            for (Set<Endpoint> endpointSet : range.values()) {
+                for (Endpoint endpoint : endpointSet) {
+                    WatchListEntry watchListEntry = endpoints.get(endpoint);
+                    if (watchListEntry != null) {
+                        carrier.discoveryRequest(endpoint, tickTime);
```

Comment:
You need to do it only if `renew.put(enpoint, watchlistEntry) == null` now you can produce 2 request for single endpoint.

Reply:
Fixed.